Dockerfile

```diff
@@ -1,6 +1,10 @@
 # Update the base image in Makefile when updating golang version. This has to
 # be pre-pulled in order to work on GCB.
+ARG ARCH
 FROM golang:1.17.1 as build
+ARG ARCH
```

Comment:
I don't think we need to move those ARGs (second ARCH, GIT_COMMIT, GIT_TAG). I think it's better to keep them before make invocation as they are only needed there. Additional benefit is that you will not need to run apt-get and go mod download every time you change git commit.

Reply:
I didn't even think twice about the impact this would have on caching, good catch. Will move these back!